source/common/upstream/load_balancer_impl.cc

```diff
@@ -695,17 +695,38 @@ const HostVector& ZoneAwareLoadBalancerBase::hostSourceToHosts(HostsSource hosts
 EdfLoadBalancerBase::EdfLoadBalancerBase(
     const PrioritySet& priority_set, const PrioritySet* local_priority_set, ClusterStats& stats,
     Runtime::Loader& runtime, Random::RandomGenerator& random,
-    const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config)
+    const envoy::config::cluster::v3::Cluster::CommonLbConfig& common_config,
+    TimeSource& time_source)
     : ZoneAwareLoadBalancerBase(priority_set, local_priority_set, stats, runtime, random,
                                 common_config),
-      seed_(random_.random()) {
+      seed_(random_.random()),
+      slow_start_window_(
+          std::chrono::milliseconds(common_config.has_slow_start_config()
+                                        ? common_config.slow_start_config().slow_start_window()
+                                        : 0) *
+          1000),
+      time_bias_runtime_(
+          common_config.has_slow_start_config() && common_config.slow_start_config().has_time_bias()
+              ? std::make_unique<Runtime::Double>(common_config.slow_start_config().time_bias(),
+                                                  runtime)
+              : nullptr),
+      time_source_(time_source),
+      hosts_in_slow_start_(
+          std::make_shared<absl::btree_set<HostSharedPtr, OrderByCreateDateDesc>>()),
+      slow_start_enabled_(slow_start_window_ > std::chrono::milliseconds(0)) {
   // We fully recompute the schedulers for a given host set here on membership change, which is
   // consistent with what other LB implementations do (e.g. thread aware).
   // The downside of a full recompute is that time complexity is O(n * log n),
   // so we will need to do better at delta tracking to scale (see
   // https://github.com/envoyproxy/envoy/issues/2874).
   priority_update_cb_ = priority_set.addPriorityUpdateCb(
       [this](uint32_t priority, const HostVector&, const HostVector&) { refresh(priority); });
+  member_update_cb_ = priority_set.addMemberUpdateCb(
```

Comment:
Is it necessary to add all hosts to the `hosts_in_slow_start_` list when it is first started?

Reply:
@zyfjeff not sure I understand the question. Could you rephrase?

Reply:
When EDS pushes all the hosts for the first time, they will be in slow mode. Is this necessary?

Reply:
@zyfjeff hosts should enter slow start mode as soon as they join the cluster (if no active HC is configured per cluster). If active HC is configured they enter slow start upon passing first HC.